[DOCS] Fix default for `http.compression` setting (#56899)
elastic/elasticsearch: docs/reference/modules/http.asciidoc

|`http.compression` |Support for compression when possible (with

> Accept-Encoding). Defaults to `true`. Please be aware if HTTPS is enabled, compression will be disabled (set to `false`) to mitigate potential security risks like the BREACH attack. If you want to enforce compress HTTPS traffic, please explicitly set this to `true`.

Reviewer: ```suggestion
Accept-Encoding). If HTTPS is enabled, defaults to `false`. Otherwise, defaults
to `true`.

Disabling compression for HTTPS mitigates potential security risks, such as a
https://en.wikipedia.org/wiki/BREACH[BREACH attack]. To compress HTTPS traffic,
you must explicitly set `http.compression` to `true`.
```
